ai-flaw-report-streamlit.py

```diff
@@ -446,34 +461,564 @@ def display_hazard_fields():
+            "policyViolation": form_data.get("Policy Violation"),
+            "severity": form_data.get("Severity"),
+            "prevalence": form_data.get("Prevalence"),
+            "impacts": form_data.get("Impacts", []),
+            "impactedStakeholders": form_data.get("Impacted Stakeholder(s)", []),
+            "riskSource": form_data.get("Risk Source", []),
+            "bountyEligibility": form_data.get("Bounty Eligibility")
+        },
+        "disclosurePlan": {
+            "disclosureIntent": form_data.get("Disclosure Intent"),
+            "disclosureTimeline": form_data.get("Disclosure Timeline"),
+            "disclosureChannels": form_data.get("Disclosure Channels", []),
+            "embargoRequest": form_data.get("Embargo Request")
+        }
+    }
+    
+    # Report-type specific sections
+    if "Real-World Events" in form_data.get("Report Types", []):
+        json_ld["realWorldEvent"] = {
+            "incidentDescription": form_data.get("Description of the Incident(s)"),
+            "implicatedSystems": form_data.get("Implicated Systems"),
+            "submitterRelationship": form_data.get("Submitter Relationship"),
+            "eventDates": form_data.get("Event Date(s)"),
+            "eventLocations": form_data.get("Event Location(s)"),
+            "experiencedHarmTypes": form_data.get("Experienced Harm Types", []),
+            "experiencedHarmSeverity": form_data.get("Experienced Harm Severity"),
+            "harmNarrative": form_data.get("Harm Narrative")
+        }
     
-    st.subheader("Recommended Recipients")
-    for rec in recommendations:
-        st.write(f"- {rec}")
+    if "Malign Actor" in form_data.get("Report Types", []):
+        json_ld["malignActor"] = {
+            "tactics": form_data.get("Tactic Select", []),
+            "impact": form_data.get("Impact", [])
+        }
     
-    # Download button
-    json_str = json.dumps(form_data, indent=4)
-    st.download_button(
-        label="Download JSON",
-        data=json_str,
-        file_name=f"ai_flaw_report_{form_data['Report ID']}.json",
-        mime="application/json"
-    )
+    if "Security Incident Report" in form_data.get("Report Types", []):
+        json_ld["securityIncident"] = {
+            "threatActorIntent": form_data.get("Threat Actor Intent"),
+            "detection": form_data.get("Detection", [])
+        }
+    
+    if "Vulnerability Report" in form_data.get("Report Types", []):
+        json_ld["vulnerability"] = {
+            "proofOfConcept": form_data.get("Proof-of-Concept Exploit")
+        }
+    
+    if "Hazard Report" in form_data.get("Report Types", []):
+        json_ld["hazard"] = {
+            "examples": form_data.get("Examples"),
+            "replicationPacket": form_data.get("Replication Packet"),
+            "statisticalArgument": form_data.get("Statistical Argument")
+        }
+    
+    return json_ld
 
+def create_db_connection():
```

Comment:
Maybe database management should get its own file too. We could create a wrapper class with the same methods, that could be either Postres or HuggingFace?